terraform-provider-constellation/internal/provider/provider_test.go

```diff
@@ -7,22 +7,58 @@ SPDX-License-Identifier: AGPL-3.0-only
 package provider
 
 import (
+	"os"
 	"testing"
 
+	"github.com/bazelbuild/rules_go/go/runfiles"
 	"github.com/hashicorp/terraform-plugin-framework/providerserver"
 	"github.com/hashicorp/terraform-plugin-go/tfprotov6"
 )
 
+const (
+	// testingConfig is a shared configuration to combine with the actual
+	// test configuration so the constellation provider is properly configured
+	// for acceptance testing.
+	testingConfig = `provider "constellation" {}`
+)
+
 // testAccProtoV6ProviderFactories are used to instantiate a provider during
 // acceptance testing. The factory function will be invoked for every Terraform
 // CLI command executed to create a provider server to which the CLI can
 // reattach.
 var testAccProtoV6ProviderFactories = map[string]func() (tfprotov6.ProviderServer, error){
-	"scaffolding": providerserver.NewProtocol6WithError(New("test")()),
+	"constellation": providerserver.NewProtocol6WithError(New("test")()),
 }
 
-func testAccPreCheck(_ *testing.T) {
-	// You can add code here to run prior to any test case execution, for example assertions
-	// about the appropriate environment variables being set are common to see in a pre-check
-	// function.
+// bazelSetTerraformBinaryPath sets the path to the Terraform binary for
+// acceptance testing when running under Bazel.
+func bazelSetTerraformBinaryPath(t *testing.T) {
+	if !runsUnderBazel() {
+		t.Skip("Skipping test as not running under Bazel.")
```

Comment:
running locally with `TF_ACC=1 go test -v ./...` also works
```suggestion
		return
```

Reply:
This will use the host's Terraform binary though, which may cause test results to diverge from what we see in the CI. IMO it makes sense to only allow the integration tests through Bazel.

Reply:
I prefer to run tests locally. The CI will run in bazel anyways and make sure it runs there?

Reply:
You can run the test locally with `bazel test //terraform-provider-constellation/internal/provider:provider_acc_test` and it will make sure to run under the same conditions as in the CI

Reply:
Can you add this to the dev docs workflow please? I still don't like it 🙈 I prefer to run a test in VS code. Could we add a warning and still allow it instead?